uber/site_sections/hotel_lottery_admin.py

```diff
@@ -31,6 +33,112 @@ def _search(session, text):
     
     return applications.filter(or_(*check_list)), ''
 
+def weight_entry(entry, hotel_room):
+    """Takes a lottery entry and a hotel room and returns an arbitrary score for how likely that applicant
+        should be to get that particular room.
+    """
+    # Higher weight increases the odds of them getting this room.
+    weight = 1.0
+    
+    # Give 10 points for being the first choice hotel, 9 points for the second, etc
+    hotel_choice_rank = 10 - entry["hotels"].index(hotel_room["id"])
+    weight += hotel_choice_rank
+    
+    # Give 10 points for being the first choice room type, 9 points for the second, etc
+    try:
+        room_type_rank = 10 - entry["room_types"].index(hotel_room["room_type"])
+        assert room_type_rank >= 0
+        weight += room_type_rank
+    except ValueError:
+        # room types are optional, so we need to figure out how much weight to give people who don't choose any
```

Comment:
When you say room types are optional, are you thinking of suite entries where they didn't opt to get into the room lottery? I think we'd have a separate run for the suite lottery, and if someone didn't opt into the room lottery they won't be included in this at all?